subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -949,6 +949,77 @@ class ArtifactTransformIntegrationTest extends AbstractHttpDependencyResolutionT
         output.contains("files: [lib.jar]")
     }
 
+    def "transforms registering the input as an output can use normalization"() {
```

Comment:
I guess this should link to the issue?

Reply:
Should the test name imply that this is about file dependencies specifically?

Reply:
This is not about file dependencies. It doesn't matter if the input is a file dependency or something else - this is about the input artifacts having the same normalized form.

Reply:
I added a link to the issue.